pkg/template/api/v1/types.go

```diff
@@ -88,3 +88,95 @@ type Parameter struct {
 	// Optional: Indicates the parameter must have a value.  Defaults to false.
 	Required bool `json:"required,omitempty" protobuf:"varint,7,opt,name=required"`
 }
+
+// TemplateInstance is ...
+type TemplateInstance struct {
+	unversioned.TypeMeta `json:",inline"`
+	// Standard object metadata.
+	kapi.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// spec is ...
+	Spec TemplateInstanceSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+
+	// status is ..
+	Status TemplateInstanceStatus `json:"status" protobuf:"bytes,3,opt,name=status"`
+}
+
+// TemplateInstanceSpec is ...
+type TemplateInstanceSpec struct {
+	// template is ...
+	Template Template `json:"template" protobuf:"bytes,1,opt,name=template"`
+	// secretref is ...
+	SecretRef kapi.LocalObjectReference `json:"secretRef" protobuf:"bytes,2,opt,name=secretRef"`
+	// impersonateuser is ...
+	ImpersonateUser string `json:"impersonateUser" protobuf:"bytes,3,opt,name=impersonateUser"`
+}
+
+// TemplateInstanceStatus is ...
+type TemplateInstanceStatus struct {
+	// conditions is ...
+	Conditions []TemplateInstanceCondition `protobuf:"bytes,1,rep,name=conditions"`
+}
+
+// TemplateInstanceCondition is ...
+type TemplateInstanceCondition struct {
+	// type is ...
+	Type TemplateInstanceConditionType `json:"type" protobuf:"bytes,1,opt,name=type,casttype=TemplateInstanceConditionType"`
+	// status is ...
+	Status kapi.ConditionStatus `json:"status" protobuf:"bytes,2,opt,name=status"`
+	// lasttransitiontime is ...
+	LastTransitionTime unversioned.Time `json:"lastTransitionTime" protobuf:"bytes,3,opt,name=lastTransitionTime"`
+	// reason is ...
+	Reason string `json:"reason" protobuf:"bytes,4,opt,name=reason"`
+	// message is ...
+	Message string `json:"message" protobuf:"bytes,5,opt,name=message"`
+}
+
+// TemplateInstanceConditionType is ...
+type TemplateInstanceConditionType string
+
+const (
+	// TemplateInstanceCreated is ...
+	TemplateInstanceCreated TemplateInstanceConditionType = "Created"
+	// TemplateInstanceFailed is ...
+	TemplateInstanceFailed TemplateInstanceConditionType = "Failed"
```

Comment:
I don't believe these are conventional according to  https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md ...